tests/pkg/expectations-builders/formation_assignment.go

```diff
@@ -0,0 +1,109 @@
+package expectations_builders
+
+import (
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"k8s.io/utils/strings/slices"
+)
+
+type FAExpectationsBuilder struct {
+	expectations map[string]map[string]fixtures.AssignmentState
+}
+
+func NewFAExpectationsBuilder() *FAExpectationsBuilder {
+	return &FAExpectationsBuilder{
+		expectations: map[string]map[string]fixtures.AssignmentState{},
+	}
+}
+
+func (b *FAExpectationsBuilder) GetExpectations() map[string]map[string]fixtures.AssignmentState {
+	return b.expectations
+}
+
+func (b *FAExpectationsBuilder) GetExpectedAssignmentsCount() int {
+	count := 0
+	for _, val := range b.expectations {
+		count += len(val)
+	}
+	return count
+}
+
+func (b *FAExpectationsBuilder) getCurrentParticipants() []string {
+	participants := make([]string, 0, len(b.expectations))
+	for participantID, _ := range b.expectations {
+		participants = append(participants, participantID)
+	}
+
+	return participants
+}
+
+func (b *FAExpectationsBuilder) WithParticipant(participantID string) *FAExpectationsBuilder {
+	if _, ok := b.expectations[participantID]; ok {
+		return b
+	}
+
+	for _, val := range b.expectations {
+		val[participantID] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}
```

Comment:
What do you think `AssignmentState` to become `AssignmentData` or something similar? Because `AssignmentState` infers only state but we have state + other data like errors, config, etc.?

Also, why on this line we set the `val[participantID] = fixtures.AssignmentState{State: "READY", Config: nil, Value: nil, Error: nil}` to READY but a few lines below "delete" it - `b.expectations[participantID] = make(map[string]fixtures.AssignmentState, len(currentParticipants)+1)`

Reply:
`expectations` structure is  `map[string]map[string]fixtures.AssignmentState`. On line 45 we are inside the for loop and `val` is an entry from `expectations` so `val`s structure is `map[string]fixtures.AssignmentStateon`.

val[participantID] =  expectations[somePreviousParticipantID][participantID] != expectations[participantID]

The flow of the method is: go throug all of the participants substructures(`map[string]fixtures.AssignmentState`) in the expectations structure(`map[string]map[string]fixtures.AssignmentState`) and add an entry(key- participantID, value the respective default state) about the new participant in each of them. Then create a new substructure(`map[string]fixtures.AssignmentState`) for the new participant in expectations, then add the entry for the loop and then add entries for all of the participants that were already added to the structure.

About the `AssignmentState` the config and the error are part of the state, it would be more correct to update the `State` field to `Status`

Reply:
👍 I overlooked the inner map(previously named var), about the `AssignmentState` Maybe I associate the "state" with the state filed in the assignment that's why I was wondering if we could use another name for the structure(`AssignmentData` or something similar) but I'm okay with the current setup as well